controllers/drplacementcontrol_controller.go

```diff
@@ -722,10 +722,14 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 		// then the DRPC should be deleted as well. The least we should do here is to clean up DPRC.
 		err := r.processDeletion(ctx, drpc, placementObj, logger)
 		if err != nil {
-			// update drpc progression only on err
 			logger.Info(fmt.Sprintf("Error in deleting DRPC: (%v)", err))
 
-			return ctrl.Result{}, r.setProgressionAndUpdate(ctx, drpc, rmn.ProgressionDeleting)
+			statusErr := r.setProgressionAndUpdate(ctx, drpc, rmn.ProgressionDeleting)
```

Comment:
This change is fine, however, given that we already log in line 725, we can just change the return to be:
```
return ctrl.Result{Requeue: true}, r.setProgressionAndUpdate(ctx, drpc, rmn.ProgressionDeleting)
```
But, we may want to keep it consistent and return an error when an error occurs. My attempt was to save the extra 5 lines of code.